src/app/utils/useMarzipano.ts

```diff
@@ -1,15 +1,21 @@
 import { useStateRef } from '@amsterdam/arm-core'
-import throttle from 'lodash.throttle'
+import debounce from 'lodash.debounce'
 import Marzipano from 'marzipano'
-import { RefObject, useEffect, useState } from 'react'
+import { RefObject, useEffect, useState, useCallback } from 'react'
 import { getOrientation } from '../../panorama/services/marzipano/marzipano'
 import { PanoParam } from '../pages/MapPage/query-params'
 
-const useMarzipano = (ref: RefObject<HTMLElement>) => {
+const useMarzipano = (
+  ref: RefObject<HTMLElement>,
+): {
+  marzipanoViewer: any
+  marzipanoViewerRef: RefObject<HTMLElement>
+  currentMarzipanoView: any
```

Comment:
According to the type definitions used in this method the type of `currentMarzipanoView` should be `PanoParam | null`.

```suggestion
  currentMarzipanoView: PanoParam | null
```

That said the `PanoParam` interface should more correctly be named `Pano` as it does not implement the `UrlParam` interface and thus is not actually a param, but rather the value produced from it.